{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9708ee28",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#count number of reads per library (prior to amplicon demultiplexing)\n",
    "SAMPLE_DIR=Samples\n",
    "COUNT=Results/n_reads_per_library.tsv\n",
    "\n",
    "{ echo -e \"Samples\\tFragmentsNumber\"; for f in `find Samples -name \"*R1_001.fastq.gz\"`; do zcat \"$f\" | echo -e \"$(basename `dirname $f | cut -d - -f 1`)\\t$(( `wc -l`/4 ))\"; done } > $COUNT"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b79c1e5",
   "metadata": {},
   "source": [
    "### 1. demultiplex amplicons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f8b485e",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p Results/Amplicons_demultiplexed\n",
    "\n",
    "for f in Samples/* \n",
    "    do \n",
    "    f=$(basename $f)\n",
    "    cutadapt \\\n",
    "    -g file:qiaseq_primers_fwd-x.fa \\\n",
    "    -G file:qiaseq_primers_rev-x.fa \\\n",
    "    --pair-adapters --no-indels -e 0.1 \\\n",
    "    --suffix ':region={name}' \\\n",
    "    -o Results/Amplicons_demultiplexed/${f}_{name}_R1.fastq \\\n",
    "    -p Results/Amplicons_demultiplexed/${f}_{name}_R2.fastq Samples/${f}*/*R1* Samples/${f}*/*R2*\n",
    "    done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f135023a",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "#move files from each region to a particular directory\n",
    "\n",
    "mkdir -p Results/Amplicons_demultiplexed/V1V2\n",
    "mkdir -p Results/Amplicons_demultiplexed/V2V3\n",
    "mkdir -p Results/Amplicons_demultiplexed/V3V4\n",
    "mkdir -p Results/Amplicons_demultiplexed/V4V5\n",
    "mkdir -p Results/Amplicons_demultiplexed/V5V7\n",
    "mkdir -p Results/Amplicons_demultiplexed/V7V9\n",
    "mkdir -p Results/Amplicons_demultiplexed/ITS\n",
    "mkdir -p Results/Amplicons_demultiplexed/unknown\n",
    "\n",
    "mv Results/Amplicons_demultiplexed/*_V1V2* Results/Amplicons_demultiplexed/V1V2\n",
    "mv Results/Amplicons_demultiplexed/*_V2V3* Results/Amplicons_demultiplexed/V2V3\n",
    "mv Results/Amplicons_demultiplexed/*_V3V4* Results/Amplicons_demultiplexed/V3V4\n",
    "mv Results/Amplicons_demultiplexed/*_V4V5* Results/Amplicons_demultiplexed/V4V5\n",
    "mv Results/Amplicons_demultiplexed/*_V5V7* Results/Amplicons_demultiplexed/V5V7\n",
    "mv Results/Amplicons_demultiplexed/*_V7V9* Results/Amplicons_demultiplexed/V7V9\n",
    "mv Results/Amplicons_demultiplexed/*_ITS* Results/Amplicons_demultiplexed/ITS\n",
    "mv Results/Amplicons_demultiplexed/*_unknown* Results/Amplicons_demultiplexed/unknown"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c6bd657",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#calculate read length for demultiplexed samples\n",
    "\n",
    "mkdir -p Results/Read_length\n",
    "\n",
    "for f in Results/Amplicons_demultiplexed/*/*\n",
    "do\n",
    "g=$(basename $f)\n",
    "cat $f| awk '{if(NR%4==2) print length($1)}' | sort -n | uniq -c > Results/Read_length/${g}_read_length.txt\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06375c9e",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#calculate read length for all samples from a given region\n",
    "#data generated here are going to contain a sum for each fastq\n",
    "#that is, the files are being concatenated instead of summed up\n",
    "#later I will sum properly using R\n",
    "\n",
    "for f in Results/Amplicons_demultiplexed/*\n",
    "do\n",
    "region=$(basename $f)\n",
    "for g in Results/Read_length/*${region}_R1*\n",
    "do\n",
    "cat $g >> Results/Read_length/${region}_R1_read_length.txt\n",
    "done\n",
    "for g in Results/Read_length/*${region}_R2*\n",
    "do\n",
    "cat $g >> Results/Read_length/${region}_R2_read_length.txt\n",
    "done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ee64bfa",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#renaming files\n",
    "\n",
    "for f in Results/Amplicons_demultiplexed/*\n",
    "do\n",
    "region=$(basename $f)\n",
    "for g in Results/Amplicons_demultiplexed/${region}/*\n",
    "do\n",
    "g=$(basename $g)\n",
    "mv ${f}/${g} ${f}/${g//\\ds*_/}\n",
    "done\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce8a1551",
   "metadata": {},
   "source": [
    "#### generate artifacts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dbde7aa6",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#a manifest indicating sequences loc must be prepared previously\n",
    "\n",
    "#programatically generating from a generic manifest a manifest for each amplicon\n",
    "for f in Results/Amplicons_demultiplexed/*\n",
    "do\n",
    "region=$(basename $f)\n",
    "sed \"s/region/$region/g\" amplicons_demux_manifest > amplicons_demux_manifest_${region}\n",
    "done\n",
    "\n",
    "#generating qiime artifacts for each amplicon\n",
    "for f in Results/Amplicons_demultiplexed/*\n",
    "do\n",
    "region=$(basename $f)\n",
    "qiime tools import \\\n",
    "  --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "  --input-path amplicons_demux_manifest_${region} \\\n",
    "  --input-format PairedEndFastqManifestPhred33V2 \\\n",
    "  --output-path Results/Amplicons_demultiplexed/${region}_demux-paired.qza\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c7fc1138",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#summarize artifacts\n",
    "\n",
    "for f in Results/Amplicons_demultiplexed/*qza\n",
    "do\n",
    "f=$(basename $f)\n",
    "f=${f%.qza}\n",
    "qiime demux summarize \\\n",
    "  --i-data Results/Amplicons_demultiplexed/${f}.qza \\\n",
    "  --o-visualization Results/Amplicons_demultiplexed/${f}.qzv\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6b6569d8",
   "metadata": {},
   "source": [
    "### 2. dada2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9e22ac1",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir Results/Dadaed\n",
    "\n",
    "trunc_len_f=( 0 0 0 0 0 0 0 )\n",
    "trunc_len_r=( 0 241 234 236 210 213 226 ) #array with different truncation parametes for R1 and R2 for each amplicon\n",
    "i=0                                 #values were chosen so that everything that comes after (3') the first instance of median quality<30 is disconsidered\n",
    "                                #order in the array: ITS, followed by 16S amplicons in the usual order\n",
    "i=0                                \n",
    "for f in Results/Amplicons_demultiplexed/[IV]*qza\n",
    "do\n",
    "region=$(basename $f)\n",
    "qiime dada2 denoise-paired \\\n",
    "    --i-demultiplexed-seqs Results/Amplicons_demultiplexed/${region} \\\n",
    "    --p-trunc-len-f ${trunc_len_f[$i]} \\\n",
    "    --p-trunc-len-r ${trunc_len_r[$i]} \\\n",
    "    --o-table Results/Dadaed/${region}_table.qza \\\n",
    "    --o-representative-sequences Results/Dadaed/${region}_rep-seqs.qza \\\n",
    "    --o-denoising-stats Results/Dadaed/${region}_stats.qza\n",
    "i=$i+1\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03c2dbda",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#summarize dada2 feature-tables\n",
    "\n",
    "for f in Results/Dadaed/*table.qza\n",
    "do\n",
    "f=$(basename $f)\n",
    "f=${f%.qza}\n",
    "qiime feature-table summarize \\\n",
    "  --i-table Results/Dadaed/${f}.qza \\\n",
    "  --o-visualization Results/Dadaed/${f}.qzv\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fef09bb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#fixing file names in dadaed repository\n",
    "\n",
    "for f in Results/Dadaed/*\n",
    "do\n",
    "mv $f ${f//\\de*_/}\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a57d6569",
   "metadata": {},
   "source": [
    "### 3. ref-based chimera removal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c479a05",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#identify chimeric seqs\n",
    "\n",
    "mkdir Results/Chimeras\n",
    "\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "qiime vsearch uchime-ref \\\n",
    "  --i-sequences Results/Dadaed/${f}_rep-seqs.qza \\\n",
    "  --i-table Results/Dadaed/${f}_table.qza \\\n",
    "  --i-reference-sequences /home/vitorheidrich/SILVA/silva-138-99-seqs.qza \\\n",
    "  --o-chimeras Results/Chimeras/${f}_chimeric-seqs.qza \\\n",
    "  --o-nonchimeras Results/Chimeras/${f}_nonchimeric-seqs.qza \\\n",
    "  --o-stats Results/Chimeras/${f}_chimera_stats.qza\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa3b9d9e",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#remove chimeric seqs from dadaed data\n",
    "\n",
    "mkdir Results/Dadaed-chimremoved\n",
    "\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "qiime feature-table filter-features \\\n",
    "  --i-table Results/Dadaed/${f}_table.qza \\\n",
    "  --m-metadata-file Results/Chimeras/${f}_nonchimeric-seqs.qza \\\n",
    "  --o-filtered-table Results/Dadaed-chimremoved/${f}_table.qza\n",
    "  \n",
    "qiime feature-table filter-seqs \\\n",
    "  --i-data Results/Dadaed/${f}_rep-seqs.qza \\\n",
    "  --m-metadata-file Results/Chimeras/${f}_nonchimeric-seqs.qza \\\n",
    "  --o-filtered-data Results/Dadaed-chimremoved/${f}_rep-seqs.qza\n",
    "  \n",
    "qiime feature-table summarize \\\n",
    "  --i-table Results/Dadaed-chimremoved/${f}_table.qza \\\n",
    "  --o-visualization Results/Dadaed-chimremoved/${f}_table.qzv\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7cb3592c",
   "metadata": {},
   "source": [
    "### 4. taxonomic assignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87109aca",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#getting silva data\n",
    "\n",
    "qiime rescript get-silva-data \\\n",
    "    --p-version '138' \\\n",
    "    --p-target 'SSURef_NR99' \\\n",
    "    --p-include-species-labels \\\n",
    "    --o-silva-sequences /home/vitorheidrich/SILVA/silva-138-ssu-nr99-seqs.qza \\\n",
    "    --o-silva-taxonomy /home/vitorheidrich/SILVA/silva-138-ssu-nr99-tax.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40a72b5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#removing low quality ref-seqs\n",
    "\n",
    "qiime rescript cull-seqs \\\n",
    "    --i-sequences /home/vitorheidrich/SILVA/silva-138-ssu-nr99-seqs.qza \\\n",
    "    --o-clean-sequences /home/vitorheidrich/SILVA/silva-138-ssu-nr99-seqs-cleaned.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7dab9f65",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#dereplicating identical seqs\n",
    "\n",
    "qiime rescript dereplicate \\\n",
    "    --i-sequences /home/vitorheidrich/SILVA/silva-138-ssu-nr99-seqs-cleaned.qza  \\\n",
    "    --i-taxa /home/vitorheidrich/SILVA/silva-138-ssu-nr99-tax.qza \\\n",
    "    --p-rank-handles 'silva' \\\n",
    "    --p-mode 'uniq' \\\n",
    "    --o-dereplicated-sequences /home/vitorheidrich/SILVA/silva-138-ssu-nr99-seqs-derep-uniq.qza \\\n",
    "    --o-dereplicated-taxa /home/vitorheidrich/SILVA/silva-138-ssu-nr99-tax-derep-uniq.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "227ab607",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#extracting specific regions from ref-seqs for each amplicon\n",
    "\n",
    "f_primer=( AGRGTTTGATYMTGGCTC GGCGNACGGGTGAGTAA CCTACGGGNGGCWGCAG GTGYCAGCMGCCGCGGTAA GGATTAGATACCCBRGTAGTC YAACGAGCGMRACCC )\n",
    "r_primer=( CTGCTGCCTYCCGTA WTTACCGCGGCTGCTGG GACTACHVGGGTATCTAATCC CCGYCAATTYMTTTRAGTTT ACGTCRTCCCCDCCTTCCTC TACGGYTACCTTGTTAYGACTT )\n",
    "i=0\n",
    "\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "qiime feature-classifier extract-reads \\\n",
    "    --i-sequences /home/vitorheidrich/SILVA/silva-138-ssu-nr99-seqs-derep-uniq.qza \\\n",
    "    --p-f-primer ${f_primer[$i]} \\\n",
    "    --p-r-primer ${r_primer[$i]} \\\n",
    "    --p-min-length 100 \\\n",
    "    --p-max-length 600 \\\n",
    "    --p-n-jobs 2 \\\n",
    "    --p-read-orientation 'forward' \\\n",
    "    --o-reads /home/vitorheidrich/SILVA/silva-138-ssu-nr99-seqs_${f}.qza\n",
    "i=$i+1\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "45244c98",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#rodado no ssh\n",
    "#training feature classifier\n",
    "\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "qiime feature-classifier fit-classifier-naive-bayes \\\n",
    "  --i-reference-reads /home/vitorheidrich/SILVA/silva-138-ssu-nr99-seqs_${f}.qza \\\n",
    "  --i-reference-taxonomy /home/vitorheidrich/SILVA/silva-138-ssu-nr99-tax-derep-uniq.qza \\\n",
    "  --o-classifier /home/vitorheidrich/SILVA/${f}_classifier.qza\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7436109",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#rodado no ssh\n",
    "#performing taxonomic assignment\n",
    "\n",
    "mkdir Results/Taxonomy\n",
    "\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "qiime feature-classifier classify-sklearn \\\n",
    "  --i-classifier /home/vitorheidrich/SILVA/${f}_classifier.qza \\\n",
    "  --i-reads Results/Dadaed-chimremoved/${f}_rep-seqs.qza \\\n",
    "  --o-classification Results/Taxonomy/${f}_taxonomy.qza\n",
    "\n",
    "qiime metadata tabulate \\\n",
    "  --m-input-file Results/Taxonomy/${f}_taxonomy.qza \\\n",
    "  --o-visualization Results/Taxonomy/${f}_taxonomy.qzv\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85b0410c",
   "metadata": {},
   "source": [
    "### 5. remove non-bacterial seqs "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "25c49681",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#remove reads unassigned (including unassigned bacteria) or non-bacterial (archaea, chloroplast, mitochondria, etc...)\n",
    "\n",
    "#mkdir Results/Dadaed-chimremoved-nonbactremoved\n",
    "\n",
    "NONBACT=Unassigned,Chloroplast,Mitochondria,Eukaryota,Archaea\n",
    "\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "qiime taxa filter-seqs \\\n",
    "  --i-sequences Results/Dadaed-chimremoved/${f}_rep-seqs.qza \\\n",
    "  --i-taxonomy Results/Taxonomy/${f}_taxonomy.qza \\\n",
    "  --p-exclude $NONBACT \\\n",
    "  --o-filtered-sequences Results/Dadaed-chimremoved-nonbactremoved/${f}_rep-seqs.qza\n",
    "\n",
    "qiime taxa filter-table\\\n",
    "  --i-table Results/Dadaed-chimremoved/${f}_table.qza \\\n",
    "  --i-taxonomy Results/Taxonomy/${f}_taxonomy.qza \\\n",
    "  --p-exclude $NONBACT \\\n",
    "  --o-filtered-table Results/Dadaed-chimremoved-nonbactremoved/${f}_table.qza\n",
    "  \n",
    "qiime taxa filter-seqs \\\n",
    "  --i-sequences Results/Dadaed-chimremoved-nonbactremoved/${f}_rep-seqs.qza \\\n",
    "  --i-taxonomy Results/Taxonomy/${f}_taxonomy.qza \\\n",
    "  --p-exclude \"d__Bacteria\" \\\n",
    "  --p-mode exact \\\n",
    "  --o-filtered-sequences Results/Dadaed-chimremoved-nonbactremoved/${f}_rep-seqs.qza\n",
    "\n",
    "qiime taxa filter-table\\\n",
    "  --i-table Results/Dadaed-chimremoved-nonbactremoved/${f}_table.qza \\\n",
    "  --i-taxonomy Results/Taxonomy/${f}_taxonomy.qza \\\n",
    "  --p-exclude \"d__Bacteria\" \\\n",
    "  --p-mode exact \\\n",
    "  --o-filtered-table Results/Dadaed-chimremoved-nonbactremoved/${f}_table.qza\n",
    "\n",
    "qiime feature-table summarize \\\n",
    "  --i-table Results/Dadaed-chimremoved-nonbactremoved/${f}_table.qza \\\n",
    "  --o-visualization Results/Dadaed-chimremoved-nonbactremoved/${f}_table.qzv\n",
    "  \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03396440",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#for V5V7, the most abundant ASV in SmartControl was assigned to Enterobacterales, \n",
    "#that is an artifact; for other amplicons the most abundant ASV was SmartControl was identified as unassigned bacteria\n",
    "#so we will filter this manually below\n",
    "\n",
    "qiime feature-table filter-seqs \\\n",
    "  --i-data Results/Dadaed-chimremoved-nonbactremoved/V5V7_rep-seqs.qza \\\n",
    "  --m-metadata-file Results/Taxonomy/V5V7_taxonomy.qza \\\n",
    "  --p-where \"[Feature ID]='d961cc92ebafe55bd35d1f411f10babb'\" \\\n",
    "  --p-exclude-ids \\\n",
    "  --o-filtered-data Results/Dadaed-chimremoved-nonbactremoved/V5V7_rep-seqs.qza\n",
    "\n",
    "qiime feature-table  filter-features\\\n",
    "  --i-table Results/Dadaed-chimremoved-nonbactremoved/V5V7_table.qza \\\n",
    "  --m-metadata-file Results/Taxonomy/V5V7_taxonomy.qza \\\n",
    "  --p-where \"[Feature ID]='d961cc92ebafe55bd35d1f411f10babb'\" \\\n",
    "  --p-exclude-ids \\\n",
    "  --o-filtered-table Results/Dadaed-chimremoved-nonbactremoved/V5V7_table.qza\n",
    "\n",
    "qiime feature-table summarize \\\n",
    "  --i-table Results/Dadaed-chimremoved-nonbactremoved/V5V7_table.qza \\\n",
    "  --o-visualization Results/Dadaed-chimremoved-nonbactremoved/V5V7_table.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "68f8e36d",
   "metadata": {},
   "source": [
    "### 6. remove contaminants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39b9e245",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#inspect contaminants of negative controls\n",
    "\n",
    "#mkdir Results/Taxa-barplots\n",
    "\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "qiime taxa barplot \\\n",
    "  --i-table Results/Dadaed-chimremoved-nonbactremoved/${f}_table.qza \\\n",
    "  --i-taxonomy Results/Taxonomy/${f}_taxonomy.qza \\\n",
    "  --o-visualization Results/Taxa-barplots/${f}_taxa-barplot_contam-inspect.qzv\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8c9784a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#the contaminants will be identified with the help of the decontam R package elsewhere"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "7eaee8be",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saved FeatureData[Sequence] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V1V2_rep-seqs.qza\n",
      "Saved FeatureTable[Frequency] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V1V2_table.qza\n",
      "Saved Visualization to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V1V2_table.qzv\n",
      "Saved FeatureData[Sequence] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V2V3_rep-seqs.qza\n",
      "Saved FeatureTable[Frequency] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V2V3_table.qza\n",
      "Saved Visualization to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V2V3_table.qzv\n",
      "Saved FeatureData[Sequence] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V3V4_rep-seqs.qza\n",
      "Saved FeatureTable[Frequency] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V3V4_table.qza\n",
      "Saved Visualization to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V3V4_table.qzv\n",
      "Saved FeatureData[Sequence] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V4V5_rep-seqs.qza\n",
      "Saved FeatureTable[Frequency] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V4V5_table.qza\n",
      "Saved Visualization to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V4V5_table.qzv\n",
      "Saved FeatureData[Sequence] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V5V7_rep-seqs.qza\n",
      "Saved FeatureTable[Frequency] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V5V7_table.qza\n",
      "Saved Visualization to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V5V7_table.qzv\n",
      "Saved FeatureData[Sequence] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V7V9_rep-seqs.qza\n",
      "Saved FeatureTable[Frequency] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V7V9_table.qza\n",
      "Saved Visualization to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V7V9_table.qzv\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#remove contaminants as identified by decontam and confirmed by manual curation\n",
    "\n",
    "#mkdir Results/Dadaed-chimremoved-nonbactremoved-contamremoved\n",
    "\n",
    "CONTAM=Pelomonas,Mycoplasma_wenyonii,Candidatus_Obscuribacter\n",
    "\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "qiime taxa filter-seqs \\\n",
    "  --i-sequences Results/Dadaed-chimremoved-nonbactremoved/${f}_rep-seqs.qza \\\n",
    "  --i-taxonomy Results/Taxonomy/${f}_taxonomy.qza \\\n",
    "  --p-exclude $CONTAM \\\n",
    "  --o-filtered-sequences Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_rep-seqs.qza\n",
    "\n",
    "qiime taxa filter-table\\\n",
    "  --i-table Results/Dadaed-chimremoved-nonbactremoved/${f}_table.qza \\\n",
    "  --i-taxonomy Results/Taxonomy/${f}_taxonomy.qza \\\n",
    "  --p-exclude $CONTAM \\\n",
    "  --o-filtered-table Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_table.qza\n",
    "\n",
    "qiime feature-table summarize \\\n",
    "  --i-table Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_table.qza \\\n",
    "  --o-visualization Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_table.qzv\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "990f2cbf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saved FeatureTable[Frequency] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V1V2_table.qza\n",
      "Saved FeatureData[Sequence] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V1V2_rep-seqs.qza\n",
      "Saved Visualization to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V1V2_table.qzv\n",
      "Saved FeatureTable[Frequency] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V2V3_table.qza\n",
      "Saved FeatureData[Sequence] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V2V3_rep-seqs.qza\n",
      "Saved Visualization to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V2V3_table.qzv\n",
      "Saved FeatureTable[Frequency] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V3V4_table.qza\n",
      "Saved FeatureData[Sequence] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V3V4_rep-seqs.qza\n",
      "Saved Visualization to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V3V4_table.qzv\n",
      "Saved FeatureTable[Frequency] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V4V5_table.qza\n",
      "Saved FeatureData[Sequence] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V4V5_rep-seqs.qza\n",
      "Saved Visualization to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V4V5_table.qzv\n",
      "Saved FeatureTable[Frequency] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V5V7_table.qza\n",
      "Saved FeatureData[Sequence] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V5V7_rep-seqs.qza\n",
      "Saved Visualization to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V5V7_table.qzv\n",
      "Saved FeatureTable[Frequency] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V7V9_table.qza\n",
      "Saved FeatureData[Sequence] to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V7V9_rep-seqs.qza\n",
      "Saved Visualization to: Results/Dadaed-chimremoved-nonbactremoved-contamremoved/V7V9_table.qzv\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#remove control samples and ASVs exclusively appearing in controls\n",
    "\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "qiime feature-table filter-samples \\\n",
    "  --i-table Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_table.qza \\\n",
    "  --m-metadata-file Results/Dadaed-chimremoved-nonbactremoved/decontam-metadata.tsv \\\n",
    "  --p-where \"[Sample_or_Control]='True Sample'\" \\\n",
    "  --p-filter-empty-features \\\n",
    "  --o-filtered-table Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_table.qza\n",
    "\n",
    "qiime feature-table filter-seqs \\\n",
    "  --i-data Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_rep-seqs.qza \\\n",
    "  --i-table Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_table.qza \\\n",
    "  --o-filtered-data Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_rep-seqs.qza\n",
    "\n",
    "qiime feature-table summarize \\\n",
    "  --i-table Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_table.qza \\\n",
    "  --o-visualization Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_table.qzv\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c9a7aae6",
   "metadata": {},
   "source": [
    "### 7. generate tree "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4bee081a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saved FeatureData[AlignedSequence] to: Results/Tree/V1V2_aligned-seqs.qza\n",
      "Saved FeatureData[AlignedSequence] to: Results/Tree/V1V2_masked-alignment.qza\n",
      "Saved Phylogeny[Unrooted] to: Results/Tree/V1V2_unrooted-tree.qza\n",
      "Saved Phylogeny[Rooted] to: Results/Tree/V1V2_rooted-tree.qza\n",
      "Saved FeatureData[AlignedSequence] to: Results/Tree/V2V3_aligned-seqs.qza\n",
      "Saved FeatureData[AlignedSequence] to: Results/Tree/V2V3_masked-alignment.qza\n",
      "Saved Phylogeny[Unrooted] to: Results/Tree/V2V3_unrooted-tree.qza\n",
      "Saved Phylogeny[Rooted] to: Results/Tree/V2V3_rooted-tree.qza\n",
      "Saved FeatureData[AlignedSequence] to: Results/Tree/V3V4_aligned-seqs.qza\n",
      "Saved FeatureData[AlignedSequence] to: Results/Tree/V3V4_masked-alignment.qza\n",
      "Saved Phylogeny[Unrooted] to: Results/Tree/V3V4_unrooted-tree.qza\n",
      "Saved Phylogeny[Rooted] to: Results/Tree/V3V4_rooted-tree.qza\n",
      "Saved FeatureData[AlignedSequence] to: Results/Tree/V4V5_aligned-seqs.qza\n",
      "Saved FeatureData[AlignedSequence] to: Results/Tree/V4V5_masked-alignment.qza\n",
      "Saved Phylogeny[Unrooted] to: Results/Tree/V4V5_unrooted-tree.qza\n",
      "Saved Phylogeny[Rooted] to: Results/Tree/V4V5_rooted-tree.qza\n",
      "Saved FeatureData[AlignedSequence] to: Results/Tree/V5V7_aligned-seqs.qza\n",
      "Saved FeatureData[AlignedSequence] to: Results/Tree/V5V7_masked-alignment.qza\n",
      "Saved Phylogeny[Unrooted] to: Results/Tree/V5V7_unrooted-tree.qza\n",
      "Saved Phylogeny[Rooted] to: Results/Tree/V5V7_rooted-tree.qza\n",
      "Saved FeatureData[AlignedSequence] to: Results/Tree/V7V9_aligned-seqs.qza\n",
      "Saved FeatureData[AlignedSequence] to: Results/Tree/V7V9_masked-alignment.qza\n",
      "Saved Phylogeny[Unrooted] to: Results/Tree/V7V9_unrooted-tree.qza\n",
      "Saved Phylogeny[Rooted] to: Results/Tree/V7V9_rooted-tree.qza\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Results/Tree’: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#generate phylogenetic trees\n",
    "\n",
    "mkdir Results/Tree\n",
    "\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "\n",
    "qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "    --i-sequences Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_rep-seqs.qza \\\n",
    "    --o-alignment Results/Tree/${f}_aligned-seqs.qza \\\n",
    "    --o-masked-alignment Results/Tree/${f}_masked-alignment.qza \\\n",
    "    --o-tree Results/Tree/${f}_unrooted-tree.qza \\\n",
    "    --o-rooted-tree Results/Tree/${f}_rooted-tree.qza \n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "839cdbfa",
   "metadata": {},
   "source": [
    "### 8. sidle"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "093a8e3d",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#ran in server DONE\n",
    "\n",
    "#prepare previously extracted regions\n",
    "\n",
    "#there is a difference between the reference database used here and the one used previously to classify sequences (stored at /home/users/vheidrich/projects_current/Metagenomics/BCG-16S/SILVA)\n",
    "#há uma diferença entre essas bases de dados e as geradas anteriormente para o feature-classifier (que estão guardadas no )\n",
    "#the one used here was generated with RESCRIPt allowing a lower number of ambiguous bases\n",
    "#specifically: rescript cull-seqs with max 3 (\"--p-num-degenerates 4\"), as Sidle recommends\n",
    "#before we used rescript cull-seqs with max 4 (\"--p-num-degenerates 5\")\n",
    "\n",
    "#Sidle-reconstructed datasets combining pairs of amplicons were built through an analogous pipeline (not shown for shortness)\n",
    "\n",
    "mkdir Results/Sidle\n",
    "\n",
    "f_primer=( AGRGTTTGATYMTGGCTC GGCGNACGGGTGAGTAA CCTACGGGNGGCWGCAG GTGYCAGCMGCCGCGGTAA GGATTAGATACCCBRGTAGTC YAACGAGCGMRACCC )\n",
    "r_primer=( CTGCTGCCTYCCGTA WTTACCGCGGCTGCTGG GACTACHVGGGTATCTAATCC CCGYCAATTYMTTTRAGTTT ACGTCRTCCCCDCCTTCCTC TACGGYTACCTTGTTAYGACTT )\n",
    "i=0\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "qiime sidle prepare-extracted-region \\\n",
    "    --i-sequences /home/vitorheidrich/SILVA/silva-138-ssu-nr99-seqs_${f}.qza \\\n",
    "    --p-region $f \\\n",
    "    --p-fwd-primer ${f_primer[$i]} \\\n",
    "    --p-rev-primer ${r_primer[$i]} \\\n",
    "    --p-trim-length 300 \\\n",
    "    --o-collapsed-kmers Results/Sidle/db_${f}_300nt-kmers.qza \\\n",
    "    --o-kmer-map Results/Sidle/db_${f}_300nt-map.qza\n",
    "i=$i+1\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "354cfe3c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saved FeatureTable[Frequency] to: Results/Sidle/V1V2_table-300nt.qza\n",
      "Saved FeatureData[Sequence] to: Results/Sidle/V1V2_rep-seqs-300nt.qza\n",
      "Saved Visualization to: Results/Sidle/V1V2_table-300nt.qzv\n",
      "Saved FeatureTable[Frequency] to: Results/Sidle/V2V3_table-300nt.qza\n",
      "Saved FeatureData[Sequence] to: Results/Sidle/V2V3_rep-seqs-300nt.qza\n",
      "Saved Visualization to: Results/Sidle/V2V3_table-300nt.qzv\n",
      "Saved FeatureTable[Frequency] to: Results/Sidle/V3V4_table-300nt.qza\n",
      "Saved FeatureData[Sequence] to: Results/Sidle/V3V4_rep-seqs-300nt.qza\n",
      "Saved Visualization to: Results/Sidle/V3V4_table-300nt.qzv\n",
      "Saved FeatureTable[Frequency] to: Results/Sidle/V4V5_table-300nt.qza\n",
      "Saved FeatureData[Sequence] to: Results/Sidle/V4V5_rep-seqs-300nt.qza\n",
      "Saved Visualization to: Results/Sidle/V4V5_table-300nt.qzv\n",
      "Saved FeatureTable[Frequency] to: Results/Sidle/V5V7_table-300nt.qza\n",
      "Saved FeatureData[Sequence] to: Results/Sidle/V5V7_rep-seqs-300nt.qza\n",
      "Saved Visualization to: Results/Sidle/V5V7_table-300nt.qzv\n",
      "Saved FeatureTable[Frequency] to: Results/Sidle/V7V9_table-300nt.qza\n",
      "Saved FeatureData[Sequence] to: Results/Sidle/V7V9_rep-seqs-300nt.qza\n",
      "Saved Visualization to: Results/Sidle/V7V9_table-300nt.qzv\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#trim reads post-hoc to the same length chosen for db trimming above\n",
    "#this choice was based on the distribution of ASV lengths (analyzed in one of the R scripts)\n",
    "\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "qiime sidle trim-dada2-posthoc \\\n",
    "    --i-table Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_table.qza \\\n",
    "    --i-representative-sequences Results/Dadaed-chimremoved-nonbactremoved-contamremoved/${f}_rep-seqs.qza \\\n",
    "    --p-trim-length 300 \\\n",
    "    --o-trimmed-table Results/Sidle/${f}_table-300nt.qza \\\n",
    "    --o-trimmed-representative-sequences Results/Sidle/${f}_rep-seqs-300nt.qza\n",
    "\n",
    "qiime feature-table summarize \\\n",
    "  --i-table Results/Sidle/${f}_table-300nt.qza\\\n",
    "  --o-visualization Results/Sidle/${f}_table-300nt.qzv\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04c57366",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#ran in server DONE\n",
    "#align regional kmers\n",
    "for f in {V1V2,V2V3,V3V4,V4V5,V5V7,V7V9}\n",
    "do\n",
    "qiime sidle align-regional-kmers \\\n",
    "    --i-kmers Results/Sidle/db_${f}_300nt-kmers.qza \\\n",
    "    --i-rep-seq Results/Sidle/${f}_rep-seqs-300nt.qza \\\n",
    "    --p-region $f \\\n",
    "    --p-n-workers 4 \\\n",
    "    --p-max-mismatch 5 \\\n",
    "    --o-regional-alignment Results/Sidle/${f}_align-map.qza\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "809e41b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#ran in server DONE\n",
    "#reconstruct database\n",
    "qiime sidle reconstruct-database \\\n",
    "    --p-region V1V2 \\\n",
    "      --i-kmer-map Results/Sidle/db_V1V2_300nt-map.qza \\\n",
    "      --i-regional-alignment Results/Sidle/V1V2_align-map.qza \\\n",
    "    --p-region V2V3 \\\n",
    "      --i-kmer-map Results/Sidle/db_V2V3_300nt-map.qza \\\n",
    "      --i-regional-alignment Results/Sidle/V2V3_align-map.qza \\\n",
    "    --p-region V3V4 \\\n",
    "      --i-kmer-map Results/Sidle/db_V3V4_300nt-map.qza \\\n",
    "      --i-regional-alignment Results/Sidle/V3V4_align-map.qza \\\n",
    "    --p-region V4V5 \\\n",
    "      --i-kmer-map Results/Sidle/db_V4V5_300nt-map.qza \\\n",
    "      --i-regional-alignment Results/Sidle/V4V5_align-map.qza \\\n",
    "    --p-region V5V7 \\\n",
    "      --i-kmer-map Results/Sidle/db_V5V7_300nt-map.qza \\\n",
    "      --i-regional-alignment Results/Sidle/V5V7_align-map.qza \\\n",
    "    --p-region V7V9 \\\n",
    "      --i-kmer-map Results/Sidle/db_V7V9_300nt-map.qza \\\n",
    "      --i-regional-alignment Results/Sidle/V7V9_align-map.qza \\\n",
    "    --o-database-summary Results/Sidle/full_summary.qza \\\n",
    "    --o-database-map Results/Sidle/full_map.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "327723a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#ran in server DONE\n",
    "#reconstruct feature table\n",
    "qiime sidle reconstruct-counts \\\n",
    "    --p-region V1V2 \\\n",
    "      --i-regional-alignment Results/Sidle/V1V2_align-map.qza \\\n",
    "      --i-regional-table Results/Sidle/V1V2_table-300nt.qza \\\n",
    "    --p-region V2V3 \\\n",
    "      --i-regional-alignment Results/Sidle/V2V3_align-map.qza \\\n",
    "      --i-regional-table Results/Sidle/V2V3_table-300nt.qza \\\n",
    "    --p-region V3V4 \\\n",
    "      --i-regional-alignment Results/Sidle/V3V4_align-map.qza \\\n",
    "      --i-regional-table Results/Sidle/V3V4_table-300nt.qza \\\n",
    "    --p-region V4V5 \\\n",
    "      --i-regional-alignment Results/Sidle/V4V5_align-map.qza \\\n",
    "      --i-regional-table Results/Sidle/V4V5_table-300nt.qza \\\n",
    "    --p-region V5V7 \\\n",
    "      --i-regional-alignment Results/Sidle/V5V7_align-map.qza \\\n",
    "      --i-regional-table Results/Sidle/V5V7_table-300nt.qza \\\n",
    "    --p-region V7V9 \\\n",
    "      --i-regional-alignment Results/Sidle/V7V9_align-map.qza \\\n",
    "      --i-regional-table Results/Sidle/V7V9_table-300nt.qza \\\n",
    "    --o-reconstructed-table Results/Sidle/full_table.qza \\\n",
    "    --p-min-counts 0 \\\n",
    "    --i-database-summary Results/Sidle/full_summary.qza \\\n",
    "    --i-database-map Results/Sidle/full_map.qza\n",
    "    \n",
    "\n",
    "qiime feature-table summarize \\\n",
    "  --i-table Results/Sidle/full_table.qza \\\n",
    "  --o-visualization Results/Sidle/full_table.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e0aa8f90",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#ran in server DONE\n",
    "#reconstruct taxonomy\n",
    "\n",
    "qiime sidle reconstruct-taxonomy \\\n",
    "    --i-reconstruction-map Results/Sidle/full_map.qza \\\n",
    "    --i-taxonomy silva-138-ssu-nr99-tax-derep-uniq.qza \\\n",
    "    --p-database 'silva' \\\n",
    "    --p-define-missing 'ignore' \\\n",
    "    --o-reconstructed-taxonomy Results/Sidle/full_taxonomy.qza\n",
    " \n",
    "qiime metadata tabulate \\\n",
    "    --m-input-file Results/Sidle/full_taxonomy.qza \\\n",
    "    --o-visualization Results/Sidle/full_taxonomy.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94339a8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%bash\n",
    "\n",
    "#NEVER USED\n",
    "\n",
    "#reconstruct phylogenetic tree\n",
    "#reconstruct consensus fragments that could not be resolved\n",
    "\n",
    "#qiime sidle reconstruct-fragment-rep-seqs \\\n",
    "#    --p-region V1V2 \\\n",
    "#    --i-regional-alignment Results/Sidle/V1V2_align-map.qza \\\n",
    "#    --p-region V2V3 \\\n",
    "#    --i-regional-alignment Results/Sidle/V2V3_align-map.qza \\\n",
    "#    --p-region V3V4 \\\n",
    "#    --i-regional-alignment Results/Sidle/V3V4_align-map.qza \\\n",
    "#    --p-region V4V5 \\\n",
    "#    --i-regional-alignment Results/Sidle/V4V5_align-map.qza \\\n",
    "#    --p-region V5V7 \\\n",
    "#    --i-regional-alignment Results/Sidle/V5V7_align-map.qza \\\n",
    "#    --p-region V7V9 \\\n",
    "#    --i-regional-alignment Results/Sidle/V7V9_align-map.qza \\\n",
    "#    --i-reconstruction-map Results/Sidle/full_map.qza \\\n",
    "#    --i-reconstruction-summary Results/Sidle/full_summary.qza \\\n",
    "#    --i-aligned-sequences database/sidle-db-aligned-sequences.qza \\ #que isso?????\n",
    "#    --o-representative-fragments Results/Sidle/full-rep-seqs-fragments.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f79f3c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%bash\n",
    "\n",
    "#NEVER USED\n",
    "\n",
    "#do the fragment insertion\n",
    "#qiime fragment-insertion sepp \\\n",
    "#    --i-representative-sequences Results/Sidle/full-rep-seqs-fragments.qza \\\n",
    "#    --i-reference-database sepp-ref-silva-138.qza \\ #this pre-built reference tree is not available\n",
    "#    --o-tree Results/Sidle/full-tree.qza \\          #i will have to either build it myself, wait for the v138\n",
    "#    --o-placements Results/Sidle/full-placements.qza#sepp ref tree release or ignore phylogenetic trees"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
