Here we provide files associated with [our paper](https://doi.org/xxxplace-holderxxx):

1. Jupyter notebook containing the bioinformatics pipeline used to process sequencing data* (`270521_BCG-16S_script.ipynb`)
2. Primers (F or R) used as inputs in cutadapt to demultiplex reads amplicon-wise (`qiaseq_primers_fwd-x.fa` and `qiaseq_primers_rev-x.fa`)
3. R script used to detect contaminants (`decontam.R`)
4. R script used to tabulate compositional distances between samples (`plotDistances.R`)
5. R script used for plotting and statistics (`Plots_paper_V3.R`)

*Raw sequencing data have been deposited in the European Nucleotide Archive (ENA) at EMBL-EBI under accession number PRJEB49145 (https://ebi.ac.uk/ena/browser/view/PRJEB49145)

If you use this data in a publication, please cite as:

Heidrich... 2022. xxx. [xxx](https://doi.org/xxxplace-holderxxx)
